### PR TITLE
TYP: add type hints to functional._dimension

### PR DESCRIPTION
**Pull request by martinfleis**
I suppose this is how typing in momepy can look? Is that an improvement? Shall we do it?

### CODE REVIEW

File: momepy/functional/_dimension.py [hunk 1]
```diff
@@ -11,7 +14,10 @@
 ]
 
 
-def volume(area, height):
+def volume(
+    area: NDArray[np.float_] | Series,
```
Comment (martinfleis) [inline momepy/functional/_dimension.py:18]: Maybe we can do with a bare `NDArray`? Since I don't know how to add the same to Series anyway.

Comment (jGaboardi) [reply momepy/functional/_dimension.py:18]: Did you try: `Series[np.float_]`

**Review by jGaboardi — APPROVED** [review]
Looks good.

**Comment by jGaboardi** [comment]
Let's merge then. Can always adjust later if we figure it out.

**Pull request merged**